### ISSUE TITLE
Uncaught TypeError: panel.item.focus is not a function

### ISSUE THREAD

**Issue by ruhatch** [ktosiek/focus-follows-mouse#3]
[Enter steps to reproduce below:]
1. Install ide-haskell
2. Mouse over the Haskell IDE panel at the bottom of the screen

**Atom Version**: 1.9.9
**System**: linux 4.7.1-1-ARCH
**Thrown From**: [focus-follows-mouse](https://github.com/ktosiek/focus-follows-mouse) package, v0.0.2
### Stack Trace

Uncaught TypeError: panel.item.focus is not a function

```
At /home/rupert/.atom/packages/focus-follows-mouse/lib/focus-follows-mouse.js:112

TypeError: panel.item.focus is not a function
    at /home/rupert/.atom/packages/focus-follows-mouse/lib/focus-follows-mouse.js:112:20
```
### Commands

```
     -0:02.1.0 ide-haskell:toggle-output (atom-text-editor.editor.is-focused.vim-mode.insert-mode)
```
### Config

``` json
{
  "core": {
    "autoHideMenuBar": true,
    "disabledPackages": [
      "wrap-guide",
      "autohide-tree-view"
    ],
    "packagesWithKeymapsDisabled": [
      "tree-view"
    ]
  }
}
```
### Installed Packages

``` coffee
# User
autocomplete-haskell, v0.6.6 (active)
autohide-tree-view, v0.25.1 (inactive)
ex-mode, v0.13.0 (inactive)
focus-follows-mouse, v0.0.2 (active)
haskell-ghc-mod, v1.17.2 (active)
ide-haskell, v1.8.1 (active)
ide-haskell-repl, v0.1.3 (active)
language-haskell, v1.9.7 (active)
nerd-treeview, v0.4.2 (active)
term3, v0.22.1 (active)
vim-mode, v0.65.1 (active)
atom-dark-syntax, v0.27.0 (inactive)
atom-dark-ui, v0.51.0 (inactive)
atom-light-syntax, v0.28.0 (inactive)
atom-light-ui, v0.43.0 (inactive)
base16-tomorrow-dark-theme, v1.1.0 (inactive)
base16-tomorrow-light-theme, v1.1.1 (inactive)
one-dark-ui, v1.3.2 (active)
one-light-ui, v1.3.2 (inactive)
one-dark-syntax, v1.2.0 (active)
one-light-syntax, v1.2.0 (inactive)
solarized-dark-syntax, v1.0.2 (inactive)
solarized-light-syntax, v1.0.2 (inactive)
about, v1.5.2 (active)
archive-view, v0.61.1 (active)
autocomplete-atom-api, v0.10.0 (active)
autocomplete-css, v0.11.1 (active)
autocomplete-html, v0.7.2 (active)
autocomplete-plus, v2.31.0 (active)
autocomplete-snippets, v1.11.0 (active)
autoflow, v0.27.0 (inactive)
autosave, v0.23.1 (active)
background-tips, v0.26.0 (active)
bookmarks, v0.41.0 (active)
bracket-matcher, v0.82.1 (active)
command-palette, v0.38.0 (inactive)
deprecation-cop, v0.54.1 (active)
dev-live-reload, v0.47.0 (active)
encoding-selector, v0.22.0 (active)
exception-reporting, v0.39.0 (active)
fuzzy-finder, v1.3.0 (active)
git-diff, v1.1.0 (active)
find-and-replace, v0.198.0 (inactive)
go-to-line, v0.31.0 (inactive)
grammar-selector, v0.48.1 (active)
image-view, v0.58.0 (active)
incompatible-packages, v0.26.1 (active)
keybinding-resolver, v0.35.0 (active)
line-ending-selector, v0.5.0 (active)
link, v0.31.1 (inactive)
markdown-preview, v0.158.1 (active)
metrics, v0.53.1 (active)
notifications, v0.64.1 (active)
open-on-github, v1.2.0 (inactive)
package-generator, v1.0.0 (inactive)
settings-view, v0.238.2 (active)
snippets, v1.0.2 (active)
spell-check, v0.67.1 (active)
status-bar, v1.4.1 (active)
styleguide, v0.47.1 (active)
symbols-view, v0.113.0 (inactive)
tabs, v0.98.1 (active)
timecop, v0.33.1 (active)
tree-view, v0.208.0 (active)
update-package-dependencies, v0.10.0 (active)
welcome, v0.34.0 (active)
whitespace, v0.32.2 (active)
wrap-guide, v0.38.1 (inactive)
language-c, v0.52.1 (active)
language-clojure, v0.21.0 (active)
language-coffee-script, v0.47.0 (active)
language-csharp, v0.12.1 (active)
language-css, v0.36.2 (active)
language-gfm, v0.86.0 (active)
language-git, v0.13.0 (active)
language-go, v0.42.0 (active)
language-html, v0.44.1 (active)
language-hyperlink, v0.16.0 (active)
language-java, v0.22.0 (active)
language-javascript, v0.119.0 (active)
language-json, v0.18.0 (active)
language-less, v0.29.3 (active)
language-make, v0.22.2 (active)
language-mustache, v0.13.0 (active)
language-objective-c, v0.15.1 (active)
language-perl, v0.35.0 (active)
language-php, v0.37.0 (active)
language-property-list, v0.8.0 (active)
language-python, v0.45.0 (active)
language-ruby, v0.68.5 (active)
language-ruby-on-rails, v0.25.0 (active)
language-sass, v0.52.0 (active)
language-shellscript, v0.22.3 (active)
language-source, v0.9.0 (active)
language-sql, v0.21.1 (active)
language-text, v0.7.1 (active)
language-todo, v0.28.0 (active)
language-toml, v0.18.0 (active)
language-xml, v0.34.8 (active)
language-yaml, v0.26.0 (active)

# Dev
No dev packages
```

**Comment by ktosiek** [comment]
Thanks, I've reproduced the problem and I'll see what can be done. Panels in Atom don't really have a concept of activation (like panes do), so it's a bit hacky.

**Comment by ruhatch** [comment]
Great, thanks!

**Comment by ktosiek** [comment]
The fix is released in 0.0.3. I'm ignoring panels that don't just use a jquery-ish item for now, as I can't find a good way of focusing them.
If you find any panels that can take focus (the ide-haskell one doesn't seem to care about it) but are not getting focused by this package - please file an issue.
